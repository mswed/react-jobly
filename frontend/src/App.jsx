import { useState } from 'react';
import { BrowserRouter, Routes, Route } from 'react-router-dom';
import './App.css';
import Homepage from './Homepage';
import Companies from './Companies';
import Company from './Company';
import Jobs from './Jobs';
import NavBar from './NavBar';
import Login from './Login';
import Signup from './Signup';
import Profile from './Profile';

function App() {
  return (
    <>
      <NavBar />
      <Routes>
        <Route path="/" element={<Homepage />} />
      </Routes>
      <Routes>
        <Route path="/companies" element={<Companies />} />
      </Routes>
      <Routes>
        <Route path="/companies/:name" element={<Company />} />
      </Routes>
      <Routes>
        <Route path="/jobs" element={<Jobs />} />
      </Routes>
      <Routes>
        <Route path="/login" element={<Login />} />
      </Routes>
      <Routes>
        <Route path="/signup" element={<Signup />} />
      </Routes>
      <Routes>
        <Route path="/profile" element={<Profile />} />
      </Routes>
    </>
  );
}

export default App;
