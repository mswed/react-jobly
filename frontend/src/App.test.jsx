import { it, expect } from 'vitest';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import App from './App';

it('renders without crashing', async () => {
  render(
    <MemoryRouter>
      <App />
    </MemoryRouter>
  );
});

it('matches snapshot', async () => {
  const { asFragment } = render(
    <MemoryRouter>
      <App />
    </MemoryRouter>
  );

  expect(asFragment()).toMatchSnapshot();
});
