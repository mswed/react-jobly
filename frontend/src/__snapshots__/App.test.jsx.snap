// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <nav>
    <a
      data-discover="true"
      href="/"
    >
      Home
    </a>
    <a
      data-discover="true"
      href="/companies"
    >
      companies
    </a>
    <a
      data-discover="true"
      href="/jobs"
    >
      Jobs
    </a>
    <a
      data-discover="true"
      href="/login"
    >
      Login
    </a>
    <a
      data-discover="true"
      href="/signup"
    >
      Signup
    </a>
    <a
      data-discover="true"
      href="/profile"
    >
      Profile
    </a>
  </nav>
  <div>
    <h1>
      Jobly
    </h1>
    <h2>
      All the jobs in one, convenient place.
    </h2>
  </div>
</DocumentFragment>
`;
